syntax = "proto3";

service PredictionService {
    rpc predict (PredictRequest) returns (PredictResponse) {}
}

/* Predict request
 * image_data: bytes of the image to process
 * all_categories (default to false): true if response should contain
 * all possible categories and false if only the selected one (Single Label)
 * all_attributes (default to false): true if response should contain
 * all possible attributes and false if only the selected ones (Multi Label)
 */
message PredictRequest {
    bytes image_data = 1;
    bool all_categories = 2;
    bool all_attributes = 3;
}

/* Predict Reply
 * predicted_categories: data associated with the predicted categories
 * predicted_attributes: data associated with the predicted attributes
 */
message PredictResponse {
    repeated Correspondence predicted_categories = 2;
    repeated Correspondence predicted_attributes = 3;
}

/* Correspondence between a label and its value */
message Correspondence {
    string label = 1;
    double value = 2;
}